src/transformers/generation_logits_process.py

```diff
@@ -240,6 +240,9 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 
 class TypicalLogitsWarper(LogitsWarper):
     def __init__(self, mass: float = 0.9, filter_value: float = -float("Inf"), min_tokens_to_keep: int = 1):
+        mass = float(mass)
+        if mass <= 0 or mass >= 1.0:
```

Comment:
Sorry `<=` is always a bit tricky with float values in Python. Could we instead do the following:
```suggestion
        if not (mass > 0 and mass < 1):
```